src/reader.cpp

```diff
@@ -4737,16 +4738,16 @@ static int read_ni(parser_t &parser, int fd, const io_chain_t &io) {
 
     // Parse into an ast and detect errors.
     auto errors = new_parse_error_list();
-    auto ast = ast::ast_t::parse(str, parse_flag_none, &*errors);
-    bool errored = ast.errored();
+    auto ast = ast_parse(str, parse_flag_none, &*errors);
```

Comment:
In general to avoid lots of cascading changes to the cpp code when replacing a type `T` with a `unique_ptr<T>` returned from rust, you can just deconstruct the unique_ptr at the call site in a one-liner, e.g.

```cpp
auto ast = std::move(*ast_parse(...));
```

but when it's a box you would need to shadow the declaration since `rust::box<T>` only lets you acquire the raw pointer and not the value, so you'd need

```cpp
auto shadow_ast = ast_parse(...);
auto &ast = *shadow_ast;
```

then the downstream code won't need to be changed.